services/failure_normalizer.py

```diff
@@ -0,0 +1,77 @@
+from typing import Dict
+
+import regex
+
+predefined_dict_of_regexes_to_match = {
+    "UUID": r"[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}",
+    "DATETIME": r"(?:19|20)[0-9]{2}-(?:(?:0?[0-9])|(?:1[012]))-(?:(?:0?[0-9])|1[0-9]|2[0-9]|3[01])T(?:0[0-9]|1[0-9]|2[0-3]):(?:[0-5][0-9]):(?:[0-5][0-9])(?:Z|(?:-(?:0[0-9]:[03]0)))?",
+    "DATETIME2": r"(?:19|20)[0-9]{2}(?:(?:0?[0-9])|(?:1[012]))(?:(?:0?[0-9])|1[0-9]|2[0-9]|3[01])T(?:0[0-9]|1[0-9]|2[0-3])(?:[0-5][0-9])(?:[0-5][0-9])(?:Z|(?:-(?:0[0-9]:[03]0)))?",
+    "DATE": r"(?:19|20)[0-9]{2}-(?:(?:0?[0-9])|(?:1[012]))-(?:(?:0?[0-9])\b|1[0-9]|2[0-9]|3[01])",
+    "TIME": r"(?:0[0-9]|1[0-9]|2[0-3]):(?:[0-5][0-9]):(?:[0-5][0-9])Z?",
+    "TIME2": r"T(?:0[0-9]|1[0-9]|2[0-3])(?:[0-5][0-9])(?:[0-5][0-9])Z?",
+    "URL": r"(?:(?:http)s?:\/\/)?\w+(?:\.\w+)+(?:(?:(?:\/*[\w\-]+\/)+(?:[\w\.]+)(:\d+:\d+)*))?",
+    "LINENO": r":\d+:\d*",
+    "HEXNUMBER": r"0?x[a-f0-9]+",
+}
+
+
+class FailureNormalizer:
+    """
+    Class for normalizing a failure message
+
+    Takes a dict of strings to strings where the key is the replacement string and the value is
+    the regex we want to match and replace occurences of and a boolean that toggles whether we
+    should ignore the predefined list of patterns in the constructor
+
+    The normalize_failure_message method on the class takes a string as a parameter
+    it will remove all occurences of a match of the regexes specified in the list
```

Comment:
[nit] 'it will remove all occurences' it's not really accurate, is it? It _replaces_ them, rather than _removing_